mmrotate/datasets/hrsc.py

```diff
@@ -0,0 +1,256 @@
+                    ]],
+                                    dtype=np.float32)
+
+                    polygon = obb2poly_np(bbox,
+                                          'le90')[0, :-1].astype(np.float32)
+                    if self.version != 'le90':
+                        bbox = np.array(
+                            poly2obb_np(polygon, self.version),
+                            dtype=np.float32)
+                    else:
+                        bbox = bbox[0, :-1]
+                    head = np.array([
+                        int(obj.find('header_x').text),
+                        int(obj.find('header_y').text)
+                    ],
+                                    dtype=np.int64)
+                except:  # noqa: E722
+                    continue
+
+                gt_bboxes.append(bbox)
+                gt_labels.append(label)
+                gt_polygons.append(polygon)
+                gt_headers.append(head)
+
+            if gt_bboxes:
+                data_info['ann']['bboxes'] = np.array(
+                    gt_bboxes, dtype=np.float32)
+                data_info['ann']['labels'] = np.array(
+                    gt_labels, dtype=np.int64)
+                data_info['ann']['polygons'] = np.array(
+                    gt_polygons, dtype=np.float32)
+                data_info['ann']['headers'] = np.array(
+                    gt_headers, dtype=np.int64)
+            else:
+                data_info['ann']['bboxes'] = np.zeros((0, 5), dtype=np.float32)
+                data_info['ann']['labels'] = np.array([], dtype=np.int64)
+                data_info['ann']['polygons'] = np.zeros((0, 8),
+                                                        dtype=np.float32)
+                data_info['ann']['headers'] = np.zeros((0, 2),
+                                                       dtype=np.float32)
+
+            if gt_polygons_ignore:
+                data_info['ann']['bboxes_ignore'] = np.array(
+                    gt_bboxes_ignore, dtype=np.float32)
+                data_info['ann']['labels_ignore'] = np.array(
+                    gt_labels_ignore, dtype=np.int64)
+                data_info['ann']['polygons_ignore'] = np.array(
+                    gt_polygons_ignore, dtype=np.float32)
+                data_info['ann']['headers_ignore'] = np.array(
+                    gt_headers_ignore, dtype=np.float32)
+            else:
+                data_info['ann']['bboxes_ignore'] = np.zeros((0, 5),
+                                                             dtype=np.float32)
+                data_info['ann']['labels_ignore'] = np.array([],
+                                                             dtype=np.int64)
+                data_info['ann']['polygons_ignore'] = np.zeros(
+                    (0, 8), dtype=np.float32)
+                data_info['ann']['headers_ignore'] = np.zeros((0, 2),
+                                                              dtype=np.float32)
```

Comment:
Seems no ignored objects are added.

Reply:
i just copied that from dota.py, all the ignored field is empty. Should I remove the if branch?

Reply:
Keep this branch is okay.****